### ISSUE TITLE
How do I get this 'v1_0_train_nus_llama.json' ?

### ISSUE THREAD

**Issue by junh0cho** [OpenDriveLab/DriveLM#20]
`finetune_data_config.yaml` reference  'v1_0_train_nus_llama.json' but how do I get this?

https://github.com/OpenDriveLab/DriveLM/blob/4e730ecf222c17f378b58649922318ce3d05b8f5/challenge/llama_adapter_v2_multimodal7b/finetune_data_config.yaml#L2

Following is my train script
```
bash exps/finetune.sh /home/junho/workspace/LLaMa2-weight ./1bcbffc43484332672092e0024a8699a6eb5f558161aebf98a7c6b1db67224d1_LORA-BIAS-7B.pth ./finetune_data_config.yaml   output/path`
```
output log
```
[18:40:41.330088] Load checkpoint ./1bcbffc43484332672092e0024a8699a6eb5f558161aebf98a7c6b1db67224d1_LORA-BIAS-7B.pth
Traceback (most recent call last):
  File "main_finetune.py", line 205, in <module>
[18:40:41.330920] read dataset config from ./finetune_data_config.yaml
    main(args)
  File "main_finetune.py", line 141, in main
    dataset_train = FinetuneDataset(args.data_config, transform=transform_train,
  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/data/dataset.py", line 52, in __init__
    meta_l = json.load(open(meta_path))
FileNotFoundError: [Errno 2] No such file or directory: 'v1_0_train_nus_llama.json'
[18:40:41.331437] DATASET CONFIG:
[18:40:41.331459] {'META': ['v1_0_train_nus_llama.json']}
Traceback (most recent call last):
  File "main_finetune.py", line 205, in <module>
    main(args)
  File "main_finetune.py", line 141, in main
    dataset_train = FinetuneDataset(args.data_config, transform=transform_train,
  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/data/dataset.py", line 52, in __init__
    meta_l = json.load(open(meta_path))
FileNotFoundError: [Errno 2] No such file or directory: 'v1_0_train_nus_llama.json'
Traceback (most recent call last):
  File "main_finetune.py", line 205, in <module>
    main(args)
  File "main_finetune.py", line 141, in main
    dataset_train = FinetuneDataset(args.data_config, transform=transform_train,
  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/data/dataset.py", line 52, in __init__
    meta_l = json.load(open(meta_path))
FileNotFoundError: [Errno 2] No such file or directory: 'v1_0_train_nus_llama.json'
Traceback (most recent call last):
  File "main_finetune.py", line 205, in <module>
    main(args)
  File "main_finetune.py", line 141, in main
    dataset_train = FinetuneDataset(args.data_config, transform=transform_train,
  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/data/dataset.py", line 52, in __init__
    meta_l = json.load(open(meta_path))
FileNotFoundError: [Errno 2] No such file or directory: 'v1_0_train_nus_llama.json'
Traceback (most recent call last):
  File "main_finetune.py", line 205, in <module>
    main(args)
  File "main_finetune.py", line 141, in main
    dataset_train = FinetuneDataset(args.data_config, transform=transform_train,
  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/data/dataset.py", line 52, in __init__
    meta_l = json.load(open(meta_path))
FileNotFoundError: [Errno 2] No such file or directory: 'v1_0_train_nus_llama.json'
Traceback (most recent call last):
  File "main_finetune.py", line 205, in <module>
    main(args)
  File "main_finetune.py", line 141, in main
    dataset_train = FinetuneDataset(args.data_config, transform=transform_train,
  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/data/dataset.py", line 52, in __init__
    meta_l = json.load(open(meta_path))
FileNotFoundError: [Errno 2] No such file or directory: 'v1_0_train_nus_llama.json'
Traceback (most recent call last):
  File "main_finetune.py", line 205, in <module>
    main(args)
  File "main_finetune.py", line 141, in main
    dataset_train = FinetuneDataset(args.data_config, transform=transform_train,
  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/data/dataset.py", line 52, in __init__
    meta_l = json.load(open(meta_path))
FileNotFoundError: [Errno 2] No such file or directory: 'v1_0_train_nus_llama.json'
Traceback (most recent call last):
  File "main_finetune.py", line 205, in <module>
    main(args)
  File "main_finetune.py", line 141, in main
    dataset_train = FinetuneDataset(args.data_config, transform=transform_train,
  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/data/dataset.py", line 52, in __init__
    meta_l = json.load(open(meta_path))
FileNotFoundError: [Errno 2] No such file or directory: 'v1_0_train_nus_llama.json'

```

**Comment by jjxjiaxue** [comment]
The script convert2llama.py will generate this file. you can modify the path in this file.

**Comment by junh0cho** [comment]
```
[10:41:01.591896] Load checkpoint ./1bcbffc43484332672092e0024a8699a6eb5f558161aebf98a7c6b1db67224d1_LORA-BIAS-7B.pth
[10:41:01.592723] read dataset config from ./finetune_data_config.yaml
[10:41:01.593261] DATASET CONFIG:                                                                                                                                                                                                                                                                                                                                                                                [10:41:01.593283] {'META': ['/home/junho/workspace/DriveLM/challenge/test_v2.json']}
[10:41:01.593444] /home/junho/workspace/DriveLM/challenge/test_v2.json: len 15
[10:41:01.593461] total length: 15
[10:41:01.602487] <data.dataset.FinetuneDataset object at 0x7fa4d40fdf40>                                                                                                                                                                                                                                                                                                                                        [10:41:01.602565] Sampler_train = <torch.utils.data.distributed.DistributedSampler object at 0x7fa4d40fdfd0>
[10:41:01.603535] Start training for 4 epochs                                                                                                                                                                                                                                                                                                                                                                    [10:41:01.606155] log_dir: ./output
Traceback (most recent call last):                                                                                                                                                                                                                                                                                                                                                                                 File "main_finetune.py", line 205, in <module>
    main(args)                                                                                                                                                                                                                                                                                                                                                                                                     File "main_finetune.py", line 173, in main
    train_stats = train_one_epoch(
  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/engine_finetune.py", line 31, in train_one_epoch
Traceback (most recent call last):
  File "main_finetune.py", line 205, in <module>                                                                                                                                                                                                                                                                                                                                                                     main(args)
  File "main_finetune.py", line 173, in main
    train_stats = train_one_epoch(
  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/engine_finetune.py", line 31, in train_one_epoch
Traceback (most recent call last):
  File "main_finetune.py", line 205, in <module>
    main(args)
  File "main_finetune.py", line 173, in main
    train_stats = train_one_epoch(
  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/engine_finetune.py", line 31, in train_one_epoch
Traceback (most recent call last):
  File "main_finetune.py", line 205, in <module>
    main(args)
  File "main_finetune.py", line 173, in main
    train_stats = train_one_epoch(
  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/engine_finetune.py", line 31, in train_one_epoch
Traceback (most recent call last):
  File "main_finetune.py", line 205, in <module>
    main(args)
  File "main_finetune.py", line 173, in main
    train_stats = train_one_epoch(
  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/engine_finetune.py", line 31, in train_one_epoch
Traceback (most recent call last):
  File "main_finetune.py", line 205, in <module>
    main(args)
  File "main_finetune.py", line 173, in main
    train_stats = train_one_epoch(
  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/engine_finetune.py", line 31, in train_one_epoch
Traceback (most recent call last):
  File "main_finetune.py", line 205, in <module>
    main(args)
  File "main_finetune.py", line 173, in main
    train_stats = train_one_epoch(
  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/engine_finetune.py", line 31, in train_one_epoch
Traceback (most recent call last):
  File "main_finetune.py", line 205, in <module>
    main(args)
  File "main_finetune.py", line 173, in main
    train_stats = train_one_epoch(
  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/engine_finetune.py", line 31, in train_one_epoch
    for data_iter_step, (examples, labels, example_mask, imgs) in enumerate(metric_logger.log_every(data_loader, print_freq, header)):
          File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/util/misc.py", line 170, in log_every
    for data_iter_step, (examples, labels, example_mask, imgs) in enumerate(metric_logger.log_every(data_loader, print_freq, header)):            for data_iter_step, (examples, labels, example_mask, imgs) in enumerate(metric_logger.log_every(data_loader, print_freq, header)):for data_iter_step, (examples, labels, example_mask, imgs) in enumerate(metric_logger.log_every(data_loader, print_freq, head
er)):
for data_iter_step, (examples, labels, example_mask, imgs) in enumerate(metric_logger.log_every(data_loader, print_freq, header)):for data_iter_step, (examples, labels, example_mask, imgs) in enumerate(metric_logger.log_every(data_loader, print_freq, header)):for data_iter_step, (examples, labels, example_mask, imgs) in enumerate(metric_logger.log_every(data_loader, print_freq, header)):

for data_iter_step, (examples, labels, example_mask, imgs) in enumerate(metric_logger.log_every(data_loader, print_freq, header)):  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/util/misc.py", line 170, in log_every



  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/util/misc.py", line 170, in log_every

  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/util/misc.py", line 170, in log_every
  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/util/misc.py", line 170, in log_every
  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/util/misc.py", line 170, in log_every
  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/util/misc.py", line 170, in log_every
  File "/home/junho/workspace/DriveLM/challenge/llama_adapter_v2_multimodal7b/util/misc.py", line 170, in log_every
            header, total_time_str, total_time / len(iterable)))    header, total_time_str, total_time / len(iterable)))header, total_time_str, total_time / len(iterable)))
        header, total_time_str, total_time / len(iterable)))

ZeroDivisionErrorheader, total_time_str, total_time / len(iterable)))header, total_time_str, total_time / len(iterable)))
    ZeroDivisionError: ZeroDivisionError

ZeroDivisionError    header, total_time_str, total_time / len(iterable))): float division by zero: ZeroDivisionError
float division by zero
float division by zero:
ZeroDivisionError
float division by zero:
float division by zero
header, total_time_str, total_time / len(iterable)))
ZeroDivisionError: float division by zero
ZeroDivisionError: float division by zero
: float division by zero
```

I have no idea why this happens.

**Comment by junh0cho** [comment]
All the respect to your effort to release the data, challenge and baseline code,
description of reproducing LLaMA-adapter-V2 on DriveLM is not intuitive.

There are too many jsons and their names are very ambiguous.
- test.json
- test_v1.json
- test_v2.json  
  - which seems to be actually training dataset of LLaMA-adapter-V2? from your answer above
  - but also used as inference ? `python demo.py --llama_dir /path/to/llama_model_weights --checkpoint /path/to/pre-trained/checkpoint.pth --data ../test_v2.json  --output ../llama-adapter-DriveLM.json`
- Difference between [llama-adapter-DriveLM.json](https://github.com/OpenDriveLab/DriveLM/blob/main/challenge/llama-adapter-DriveLM.json) and [test_v1.json](https://github.com/OpenDriveLab/DriveLM/blob/main/challenge/test_v1.json)

I am also confused why would we should prepare the [nuscenes](https://www.nuscenes.org/) dataset which can refer to [BEVFormer](https://github.com/fundamentalvision/BEVFormer/blob/master/docs/prepare_dataset.md).

```
data/nuscenes                                    
├── samples                           
│   ├── n015-2018-11-21-19-58-31+0800__CAM_FRONT_LEFT__1542801707504844.jpg 
│   ├── n015-2018-11-21-19-58-31+0800__CAM_FRONT_LEFT__1542801708004844.jpg
``` 
as description.
Shouldn't we have `CAM_BACK  CAM_BACK_LEFT  CAM_BACK_RIGHT  CAM_FRONT  CAM_FRONT_LEFT  CAM_FRONT_RIGHT` as https://github.com/OpenDriveLab/DriveLM/tree/main/challenge/data/samples?

**Comment by jjxjiaxue** [comment]
> **2. Which adapter model?**
> 
> refer to | The pre-trained checkpoint can be downloaded in [ckpts](https://github.com/OpenGVLab/LLaMA-Adapter/releases/tag/v.2.0.0).
> 
> [1bcbffc43484332672092e0024a8699a6eb5f558161aebf98a7c6b1db67224d1_LORA-BIAS-7B.pth](https://github.com/OpenGVLab/LLaMA-Adapter/releases/download/v.2.0.0/1bcbffc43484332672092e0024a8699a6eb5f558161aebf98a7c6b1db67224d1_LORA-BIAS-7B.pth) [5088aeb63a89746b90bcfd5cb819e1c7411b2771b267c6d131ce73e250a8abf0_CAPTION-7B.pth](https://github.com/OpenGVLab/LLaMA-Adapter/releases/download/v.2.0.0/5088aeb63a89746b90bcfd5cb819e1c7411b2771b267c6d131ce73e250a8abf0_CAPTION-7B.pth) [7fa55208379faf2dd862565284101b0e4a2a72114d6490a95e432cf9d9b6c813_BIAS-7B.pth](https://github.com/OpenGVLab/LLaMA-Adapter/releases/download/v.2.0.0/7fa55208379faf2dd862565284101b0e4a2a72114d6490a95e432cf9d9b6c813_BIAS-7B.pth)
> 
> what do we use among above three of them?

You can use any of these three checkpoints; they only differ in their pretraining methods. For more information you can refer to: LLaMA-Adapter/llama_adapter_v2_multimodal7b at main · OpenGVLab/LLaMA-Adapter

**Comment by jjxjiaxue** [comment]
> All the respect to your effort to release the data, challenge and baseline code, description of reproducing LLaMA-adapter-V2 on DriveLM is not intuitive.
> 
> **1. There are too many jsons and their names are very ambiguous.**
> 
> * test.json
> * test_v1.json
> * test_v2.json
>   
>   * which seems to be actually training dataset of LLaMA-adapter-V2? from your answer above
>   * but also used as inference ? `python demo.py --llama_dir /path/to/llama_model_weights --checkpoint /path/to/pre-trained/checkpoint.pth --data ../test_v2.json  --output ../llama-adapter-DriveLM.json`
> * Difference between [llama-adapter-DriveLM.json](https://github.com/OpenDriveLab/DriveLM/blob/main/challenge/llama-adapter-DriveLM.json) and [test_v1.json](https://github.com/OpenDriveLab/DriveLM/blob/main/challenge/test_v1.json)

- for the training dataset of LLaMA-adapter-V2, you can refer to https://github.com/OpenDriveLab/DriveLM/blob/main/challenge/README.md#:~:text=We%20use%20llama%2Dadapter%20v2%20as%20our%20baseline.%20If%20you%20want%20to%20convert%20data%20into%20llama%2Dadapter%20format%3A , the output file of this script is the training dataset of LLaMA-adapter-V2.
- llama-adapter-DriveLM.json is the llama output file (https://github.com/OpenDriveLab/DriveLM/tree/main/challenge#:~:text=Then%20we%20will%20get%20the%20llama%2Dadapter%2DDriveLM.json%2C%20which%20are%20the%20predicted%20answers%20used%20for%20evaluation%20purposes.), and test_v2.json contains the ground truth. test_v1.json contains the same content with test_v2.json, but with different format.

**Comment by jjxjiaxue** [comment]
> **3. Nuscenes preparation** 
> ```
> data/nuscenes                                    
> ├── samples                           
> │   ├── n015-2018-11-21-19-58-31+0800__CAM_FRONT_LEFT__1542801707504844.jpg 
> │   ├── n015-2018-11-21-19-58-31+0800__CAM_FRONT_LEFT__1542801708004844.jpg
> ```

We apologize for the previous inaccurate description of this part. The correct file structure is as follow: 
 ```
 data/nuscenes                                    
 ├── samples     
 │   ├── CAM_FRONT_LEFT        
 │    │   ├── n015-2018-11-21-19-58-31+0800__CAM_FRONT_LEFT__1542801707504844.jpg 
 │    │   ├── n015-2018-11-21-19-58-31+0800__CAM_FRONT_LEFT__1542801708004844.jpg
 ```

**Comment by junh0cho** [comment]
First, this error mssing image occurs, which is not in new git pull folder.

```
[17:00:34.148994] Sampler_train = <torch.utils.data.distributed.DistributedSampler object at 0x7f1d54686fa0>
[17:00:34.149979] Start training for 4 epochs
[17:00:34.152465] log_dir: ./output
[ WARN:0@25.295] global loadsave.cpp:248 findDecoder imread_('data/nuscenes/samples/CAM_BACK_RIGHT/n008-2018-09-18-13-10-39-0400__CAM_BACK_RIGHT__1537291002278113.jpg'): can't open/read file: check file path/integrity
[ WARN:0@25.301] global loadsave.cpp:248 findDecoder imread_('data/nuscenes/samples/CAM_BACK_RIGHT/n008-2018-09-18-13-10-39-0400__CAM_BACK_RIGHT__1537291002278113.jpg'): can't open/read file: check file path/integrity
```

Therefore, I use nuscenes subset [Google Drive](https://drive.google.com/file/d/1DeosPGYeM2gXSChjMODGsQChZyYDmaUz/view?usp=sharing), and missing file is no longer problem. Maybe my json is already old from continuous update on this repo.
Then, I face https://github.com/OpenDriveLab/DriveLM/issues/20#issuecomment-1975515677 this agian.
Therefore, I somehow success after reducing batchsize 4->2, and nproc_per_node 8->4

```
[17:08:10.272124] Epoch: [0]  [0/2]  eta: 0:00:07  lr: 0.000000  closs: 10.4193 (10.4193)  mloss: 10.4193 (10.4193)  time: 3.8135  data: 1.0478  max mem: 22709
[17:08:10.906954] Epoch: [0]  [1/2]  eta: 0:00:02  lr: 0.000016  closs: 8.4814 (9.4503)  mloss: 8.4814 (9.4503)  time: 2.2238  data: 0.5240  max mem: 22717
[17:08:11.021063] Epoch: [0] Total time: 0:00:04 (2.2820 s / it)
[17:08:11.021865] Averaged stats: lr: 0.000016  closs: 8.4814 (9.2944)  mloss: 8.4814 (9.2944)
[17:08:26.618807] log_dir: ./output
[17:08:27.934318] Epoch: [1]  [0/2]  eta: 0:00:02  lr: 0.000031  closs: 10.6781 (10.6781)  mloss: 10.6781 (10.6781)  time: 1.3147  data: 0.6851  max mem: 22717
[17:08:28.571727] Epoch: [1]  [1/2]  eta: 0:00:00  lr: 0.000029  closs: 8.3587 (9.5184)  mloss: 8.3587 (9.5184)  time: 0.9757  data: 0.3426  max mem: 22717
[17:08:28.674636] Epoch: [1] Total time: 0:00:02 (1.0279 s / it)
[17:08:28.675487] Averaged stats: lr: 0.000029  closs: 8.3587 (9.4675)  mloss: 8.3587 (9.4675)
[17:08:28.680785] log_dir: ./output
[17:08:30.046981] Epoch: [2]  [0/2]  eta: 0:00:02  lr: 0.000023  closs: 10.8141 (10.8141)  mloss: 10.8141 (10.8141)  time: 1.3655  data: 0.7135  max mem: 22717
[17:08:30.682769] Epoch: [2]  [1/2]  eta: 0:00:01  lr: 0.000016  closs: 9.3862 (10.1001)  mloss: 9.3862 (10.1001)  time: 1.0003  data: 0.3568  max mem: 22727
[17:08:30.766742] Epoch: [2] Total time: 0:00:02 (1.0429 s / it)
[17:08:30.776644] Averaged stats: lr: 0.000016  closs: 9.3862 (9.3639)  mloss: 9.3862 (9.3639)
[17:08:30.781686] log_dir: ./output
[17:08:32.128286] Epoch: [3]  [0/2]  eta: 0:00:02  lr: 0.000008  closs: 10.7437 (10.7437)  mloss: 10.7437 (10.7437)  time: 1.3458  data: 0.7125  max mem: 22727
[17:08:32.768627] Epoch: [3]  [1/2]  eta: 0:00:00  lr: 0.000002  closs: 8.1265 (9.4351)  mloss: 8.1265 (9.4351)  time: 0.9928  data: 0.3563  max mem: 22727
[17:08:32.863672] Epoch: [3] Total time: 0:00:02 (1.0409 s / it)
[17:08:32.864955] Averaged stats: lr: 0.000002  closs: 8.1265 (9.1765)  mloss: 8.1265 (9.1765)
[17:08:48.412129] Training time 0:00:41
```

I better use full train size rather then small samples like you provided in the train_demo.

**Comment by junh0cho** [comment]
closing as my issue has been addressed. Thanks!

**Comment by DevLinyan** [comment]
Thank you for your suggestions; we have implemented the following changes:
1. We have adjusted the names of the JSON files to ensure they are more closely aligned with their respective contents.
2. Users now have the flexibility to utilize any of the pretrained models available in llama-adapter-v2.
3. The file path has been corrected and updated accordingly:
```bash
data/nuscenes                                    
├── samples     
│   ├── CAM_FRONT_LEFT        
│   │   ├── n015-2018-11-21-19-58-31+0800__CAM_FRONT_LEFT__1542801707504844.jpg 
│   │   ├── n015-2018-11-21-19-58-31+0800__CAM_FRONT_LEFT__1542801708004844.jpg
```
4. To ensure the execution of a complete batch, we have expanded the dataset in the demo data.
